core/src/main/java/org/apache/iceberg/view/ViewMetadata.java

```diff
@@ -124,162 +136,170 @@ static Builder buildFrom(ViewMetadata base) {
 
   class Builder {
     private static final int LAST_ADDED = -1;
+    private final List<ViewVersion> versions;
+    private final List<Schema> schemas;
+    private final List<ViewHistoryEntry> history;
+    private final Map<String, String> properties;
+    private final List<MetadataUpdate> changes;
     private int formatVersion = DEFAULT_VIEW_FORMAT_VERSION;
-    private String location;
-    private List<Schema> schemas = Lists.newArrayList();
     private int currentVersionId;
-    private Integer lastAddedVersionId;
-    private Integer lastAddedSchemaId;
```

Comment:
Tracking the last added schema ID was not necessary because we never set the schema ID directly. Now the schema is added at the same time a new version is added in `setCurrentVersion(ViewVersion, Schema)`.